### PR TITLE
[5.1] Fix multibyte string functions

### PR DESCRIPTION
**Pull request by hz61p1**

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -283,9 +283,9 @@ public static function equals($knownString, $userInput)
             return hash_equals($knownString, $userInput);
         }
 
-        $knownLength = mb_strlen($knownString, '8bit');
+        $knownLength = static::length($knownString, '8bit');
```
Comment (vlakoff) [inline src/Illuminate/Support/Str.php]: Break here because `Str::length` has only one parameter. Same at line 288.

Comment (vlakoff) [inline src/Illuminate/Support/Str.php]: By the way, this revealed there is no tests for `Str::equals` ;)

Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: just revert that change there please

Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: this is a security critical function, and does not need changing

**Comment by vlakoff** [comment]
Some of the changes here may be relevant, but you should provide unit tests demonstrating their usefulness.

File: src/Illuminate/Support/Str.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php]: Ping @odoveiz. Please revert this broken change.

**Comment by taylorotwell** [comment]
I merged this by hand.

**Comment by vlakoff** [comment]
Still, there should be tests where the mb* functions make a difference.

**Pull request closed without merging**